src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private void addCheck(int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Your index was " + index
+                    + ", but size is " + size);
+        }
+    }
+
+    private void indexCheck(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Your index was " + index
+                    + ", but size is " + size);
+        }
+    }
+
+    private Node<T> findNodeByValue(T value) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (value == node.item || value != null
+                    && value.equals(node.item)) {
+                return node;
+            }
+            node = node.next;
+        }
+        throw new RuntimeException("Element " + value + " didn't find");
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        indexCheck(index);
+        int part = size % 2 == 0 ? size / 2 : size / 2 + 1;
+        if (index < part) {
+            Node<T> node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+            return node;
+        }
+        Node<T> node = tail;
+        for (int i = 0; i < size - index - 1; i++) {
+            node = node.prev;
+        }
+        return node;
+    }
+
+    private void linkBefore(T element, Node<T> next) {
+        Node<T> prev = next.prev;
+        Node<T> newNode = new Node<>(prev, element, next);
+        next.prev = newNode;
+        if (prev == null) {
+            head = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        size++;
+    }
+
+    private void setFirstValue(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        head = newNode;
+        tail = newNode;
+        size++;
+    }
+
+    private void linkLast(T element) {
+        Node<T> newNode = new Node<>(tail, element, null);
+        tail.next = newNode;
+        tail = newNode;
+        size++;
+    }
+
+    private void unlink(Node<T> node) {
+        size--;
+        if (node == head && node == tail) {
```

Comment:
replace  with if (seze == 1)

Reply:
There, everything is checked through the tail and head variables, so if I change 1 if(), then I need to change the logic of the method